### PR TITLE
Handle OS termination signals.

### PR DESCRIPTION
**Pull request by jlesage**
This pull request allows dupeGuru to terminate gracefully when receiving an OS signal.

### CODE REVIEW

File: run.py [hunk 1]
```diff
@@ -30,10 +47,18 @@ def main():
     lang = settings.value('Language')
     locale_folder = op.join(BASE_PATH, 'locale')
     install_gettext_trans_under_qt(locale_folder, lang)
+    # Handle OS signals
+    setUpSignals()
+    # Let the Python interpreter run every 500ms to handle signals.
+    from PyQt5.QtCore import QTimer
+    timer = QTimer()
```
Comment (hsoft) [inline run.py]: I don't understand this. What happens if we don't have this timer? Qt's runloop doesn't let signal handling work properly?

Comment (jlesage) [inline run.py]: Exactly.  Without this, Python cannot handle signals while the Qt event loop is running.

**Comment by hsoft** [comment]
Thanks for this PR. Although I'm not sure I understand the actual use case, it's usually a good idea to gracefully handle those things. I'll give it a try soon.

File: run.py (hunk 1 above)
Comment (hsoft) [inline run.py]: Could we please have a comment that mentions the Qt's runloop then? The role of this piece of code is hard to grasp without the context of the PR.

**Comment by jlesage** [comment]
Here is some context about the reason I did this fix:

dupeGuru saves some stuff only during its shutdown routine.  If the application is terminated via signals, the shutdown routine is not called and thus not everything is saved to disk.

I created a docker container to run dupeGuru.  With this fix, we can stop and restart the container and make dupeGuru shutdown gracefully.  In other words, dupeGuru properly saves everything when the container is restarted/stopped.

**Comment by jlesage** [comment]
Also, here is a source where I took some inspiration:
https://stackoverflow.com/questions/4938723/what-is-the-correct-way-to-make-my-pyqt-application-quit-when-killed-from-the-co

**Comment by hsoft** [comment]
Merging, thanks!

**Pull request merged**